tests/reftests/install-formula.test

```diff
@@ -0,0 +1,226 @@
+Everything as up-to-date as possible (run with --verbose to show unavailable upgrades).
+
+The following packages are not being upgraded because the new versions conflict with other installed packages:
+  - ocaml.4.10.0
+    -- ocaml-freestanding.0.4.5 is installed and requires ocaml (>= 4.05.0 & < 4.09.0)
+    -- ocaml-src.4.08.0 is installed and requires ocaml = 4.08.0
+However, you may "opam upgrade" these packages explicitly, which will ask permission to downgrade or uninstall the conflicting packages.
+Nothing to do.
+### opam upgrade mirage-solo5 --formula '"mirage-no-solo5"' --best-effort
+The following actions would be faked:
+  - remove  mirage-solo5    0.5.0
+  - install mirage-no-solo5 1
+===== 1 to install | 1 to remove =====
+### OPAMSHOW=0 opam switch create test --empty
+### opam install --formula '"ocaml-base-compiler" | "ocaml-system"'
+The following actions would be faked:
+  - install base-bigarray       base
+  - install ocaml-base-compiler 4.08.0
+  - install base-threads        base
+  - install base-unix           base
+  - install ocaml-config        1
+  - install ocaml               4.08.0
+===== 6 to install =====
+### opam install ocaml --formula '"ocaml-variants" | "ocaml-system"'
+The following actions would be faked:
+  - install base-bigarray base
+  - install base-threads  base
+  - install base-unix     base
+  - install ocaml-system  4.08.0
+  - install ocaml-config  1      [required by ocaml]
+  - install ocaml         4.08.0
+===== 6 to install =====
+### opam install ocaml.4.07.0 --formula '"ocaml-base-compiler" | "ocaml-system"'
+The following actions would be faked:
+  - install base-bigarray       base
+  - install ocaml-base-compiler 4.07.0
+  - install base-threads        base
+  - install base-unix           base
+  - install ocaml-config        1      [required by ocaml]
+  - install ocaml               4.07.0
+===== 6 to install =====
+### opam install ocp-index.1.0.0 --formula '"ocaml-system" | "ocaml-variants"'
+The following actions would be faked:
+  - install base-bigarray   base
+  - install base-threads    base
+  - install ocaml-variants  4.01.0+profile
+  - install base-unix       base
+  - install conf-m4         1              [required by ocamlfind]
+  - install ocaml-config    1              [required by ocaml]
+  - install ocaml           4.01.0         [required by ocp-index]
+  - install ocamlfind       1.8.0          [required by ocp-build]
+  - install base-ocamlbuild base           [required by ocamlbuild]
+  - install ocp-build       1.99.19-beta   [required by ocp-index]
+  - install ocamlbuild      0              [required by cmdliner]
+  - install cmdliner        0.9.7          [required by ocp-index]
+  - install ocp-indent      1.4.0          [required by ocp-index]
+  - install ocp-index       1.0.0
+===== 14 to install =====
+### OPAMNODEPEXTS=1
+### opam install genet --formula '["mysql" ("conf-mariadb" & "mariadb" | "conf-postgresql")]'
```

Comment:
what’s with the brackets here? I’ve never seen this syntax before and it’s not in the documentation

Reply:
it's just the same syntax as the `depends:` field

Reply:
ah i see